plugins/outputs/azure_monitor_logs/azure_monitor_logs.go

```diff
@@ -0,0 +1,253 @@
+	if err != nil {
+		return err
+	} else if !match {
+		return fmt.Errorf("namespace_prefix contains invalid characters")
+	}
+
+	return nil
+}
+
+// Connect validates connectivity
+func (a *AzLogAnalytics) Connect() error {
+
+	err := a.Init()
+	if err != nil {
+		return err
+	}
+
+	client, err := a.createClient(context.Background())
+	if err != nil {
+		return err
+	}
+
+	a.client = client
+	a.serviceURL = fmt.Sprintf(baseURL, a.CustomerID)
+
+	return nil
+}
+
+// Close shuts down an any active connections
+func (a *AzLogAnalytics) Close() error {
+	a.client.CloseIdleConnections()
+
+	return nil
+}
+
+// Description provides a description of the plugin
+func (a *AzLogAnalytics) Description() string {
+	return "A plugin that can transmit metrics to Azure Log Analytics"
+}
+
+// SampleConfig provides a sample configuration for the plugin
+func (a *AzLogAnalytics) SampleConfig() string {
+	return sampleConfig
+}
+
+// Write writes metrics to the remote endpoint
+func (a *AzLogAnalytics) Write(metrics []telegraf.Metric) error {
+
+	objects := make(map[string][]interface{}, len(metrics))
+	for _, metric := range metrics {
+		m, name := a.createObject(metric)
+		objects[name] = append(objects[name], m)
+	}
+
+	for name, data := range objects {
+		serialized, err := json.Marshal(data)
+		if err != nil {
+			return err
+		}
+
+		if err := a.write(name, serialized); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (a *AzLogAnalytics) write(logType string, reqBody []byte) error {
+
+	dateString := time.Now().UTC().Format(time.RFC1123)
+	dateString = strings.Replace(dateString, "UTC", "GMT", -1)
+
+	stringToHash := httpMethod + "\n" + 
+					strconv.Itoa(utf8.RuneCount(reqBody)) + "\n" +
+					contentType + "\n" + "x-ms-date:" +
+					dateString + "\n/api/logs"
+
+	signature, err := a.buildSignature(stringToHash)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest(httpMethod, a.serviceURL, bytes.NewReader(reqBody))
+	if err != nil {
+		return err
+	}
+
+	tableName := a.NamespacePrefix + underscoreToCaml(logType)
+
+	req.Header.Set("User-Agent", "Telegraf/" + internal.ProductToken())
+	req.Header.Add("Log-Type", tableName)
+	req.Header.Add("Authorization", signature)
+	req.Header.Add("Content-Type", contentType)
+	req.Header.Add("x-ms-date", dateString)
+	req.Header.Add("time-generated-field", timeGeneratedFieldName)
+
+	resp, err := a.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	_, err = ioutil.ReadAll(resp.Body)
+
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		return fmt.Errorf(
+			"when writing to [%s] received status code: %d",
+			a.serviceURL,
+			resp.StatusCode)
+	}
+
+	return nil
+}
+
+func (a *AzLogAnalytics) createClient(ctx context.Context) (*http.Client, error) {
+	client := &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+		},
+		Timeout: a.ClientTimeout.Duration,
+	}
+
+	return client, nil
+}
```

Comment:
If you _really_ want this as an own function, please only return the client as this cannot produce an error.